app/services/subscriptions/create_service.rb

```diff
@@ -42,21 +46,29 @@ def process_create
       return result.fail!('missing_argument', 'unable to find customer') unless current_customer
       return result.fail!('missing_argument', 'plan does not exists') unless current_plan
 
+      if currency_missmatch?(current_customer&.active_subscription&.plan, current_plan)
```

Comment:
Shouldn't we use `@current_subscription` instead of `current_customer&.active_subscription` here?

Reply:
we will only have @current_subscription when there is upgrade or downgrade. However, there is condition that all subscriptions assigned to one customer should have the same currency. so that's why I'm fetching the last active one (since this cover upgrade/downgrade + adding new plan to customer)